### ISSUE TITLE
Can't run tests

### ISSUE THREAD

**Issue by kiafaldorius** [Compass/compass#1431]
I can't run the unit tests. Features run ok.

This is on the stable branch.

Why is Gemfile.lock gitignored? It's meant to handle this exact case of gem version differences.

```
[work@archie-work compass]$ bundle list
Gems included by the bundle:
  * actionmailer (3.2.15)
  * actionpack (3.2.15)
  * activemodel (3.2.15)
  * activerecord (3.2.15)
  * activeresource (3.2.15)
  * activesupport (3.2.15)
  * arel (3.0.2)
  * builder (3.0.4)
  * bundler (1.3.5)
  * celluloid (0.15.2)
  * chunky_png (1.2.9)
  * coderay (1.0.9)
  * compass (0.12.2.1bf08c9 1bf08c9)
  * compass-validator (3.0.1)
  * css_parser (1.0.1)
  * cucumber (1.1.9)
  * diff-lcs (1.1.3)
  * erubis (2.7.0)
  * ffi (1.9.3)
  * formatador (0.2.4)
  * fssm (0.2.10)
  * gherkin (2.9.3)
  * guard (2.2.2)
  * guard-cucumber (0.7.5)
  * guard-test (2.0.2)
  * haml (3.1.8)
  * hike (1.2.3)
  * i18n (0.6.5)
  * journey (1.0.4)
  * json (1.8.1)
  * listen (2.2.0)
  * lumberjack (1.0.4)
  * mail (2.5.4)
  * metaclass (0.0.1)
  * method_source (0.8.2)
  * mime-types (1.25)
  * mocha (0.14.0)
  * multi_json (1.8.2)
  * packager (0.1.0)
  * polyglot (0.3.3)
  * pry (0.9.12.2)
  * rack (1.4.5)
  * rack-cache (1.2)
  * rack-ssl (1.3.3)
  * rack-test (0.6.2)
  * rails (3.2.15)
  * railties (3.2.15)
  * rake (0.9.6)
  * rb-fsevent (0.9.3)
  * rb-inotify (0.9.2)
  * rdoc (3.12.2)
  * rspec (2.0.1)
  * rspec-core (2.0.1)
  * rspec-expectations (2.0.1)
  * rspec-mocks (2.0.1)
  * ruby_gntp (0.3.4)
  * rubyzip (1.0.0)
  * sass (3.2.12)
  * slop (3.4.6)
  * sprockets (2.2.2)
  * term-ansicolor (1.2.2)
  * test-unit (2.5.5)
  * thor (0.18.1)
  * tilt (1.4.1)
  * timers (1.1.0)
  * tins (0.12.0)
  * treetop (1.4.15)
  * tzinfo (0.3.38)
```

```
[work@archie-work compass]$ bundle exec rake test --trace
WARNING: cannot load such file -- rcov/rcovtask
** Invoke test (first_time)
** Execute test
/home/work/.rvm/rubies/ruby-2.0.0-p247/bin/ruby -I"lib:lib:test" -I"/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0" "/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb" "test/integrations/compass_test.rb" "test/integrations/sprites_test.rb" "test/units/actions_test.rb" "test/units/command_line_test.rb" "test/units/compass_module_test.rb" "test/units/compass_png_test.rb" "test/units/compiler_test.rb" "test/units/configuration_test.rb" "test/units/regressions_test.rb" "test/units/sass_extensions_test.rb" "test/units/sprites/engine_test.rb" "test/units/sprites/image_row_test.rb" "test/units/sprites/image_test.rb" "test/units/sprites/importer_test.rb" "test/units/sprites/layout_test.rb" "test/units/sprites/row_fitter_test.rb" "test/units/sprites/sprite_command_test.rb" "test/units/sprites/sprite_map_test.rb" 

*** Mocha deprecation warning: Change `require 'mocha'` to `require 'mocha/setup'`.

/home/work/Fork-Projects/compass/test/helpers/test_case.rb:42:in `setup': wrong number of arguments (2 for 0) (ArgumentError)
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/integration/test_unit/adapter.rb:27:in `included'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/integration/test_unit.rb:43:in `include'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/integration/test_unit.rb:43:in `activate'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/integration.rb:8:in `map'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/integration.rb:8:in `activate'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/setup.rb:6:in `activate'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha/setup.rb:10:in `<top (required)>'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha.rb:7:in `require'
    from /home/work/.rvm/gems/ruby-2.0.0-p247/gems/mocha-0.14.0/lib/mocha.rb:7:in `<top (required)>'
    from /home/work/Fork-Projects/compass/test/units/sprites/image_test.rb:2:in `require'
    from /home/work/Fork-Projects/compass/test/units/sprites/image_test.rb:2:in `<top (required)>'
    from /home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb:15:in `require'
    from /home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb:15:in `block in <main>'
    from /home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb:4:in `select'
    from /home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb:4:in `<main>'
rake aborted!
Command failed with status (1): [ruby -I"lib:lib:test" -I"/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0" "/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/rake_test_loader.rb" "test/integrations/compass_test.rb" "test/integrations/sprites_test.rb" "test/units/actions_test.rb" "test/units/command_line_test.rb" "test/units/compass_module_test.rb" "test/units/compass_png_test.rb" "test/units/compiler_test.rb" "test/units/configuration_test.rb" "test/units/regressions_test.rb" "test/units/sass_extensions_test.rb" "test/units/sprites/engine_test.rb" "test/units/sprites/image_row_test.rb" "test/units/sprites/image_test.rb" "test/units/sprites/importer_test.rb" "test/units/sprites/layout_test.rb" "test/units/sprites/row_fitter_test.rb" "test/units/sprites/sprite_command_test.rb" "test/units/sprites/sprite_map_test.rb" ]
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/testtask.rb:104:in `block (3 levels) in define'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils.rb:45:in `call'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils.rb:45:in `sh'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils_ext.rb:40:in `sh'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils.rb:82:in `ruby'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils_ext.rb:40:in `ruby'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/testtask.rb:100:in `block (2 levels) in define'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/file_utils_ext.rb:61:in `verbose'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/testtask.rb:98:in `block in define'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:228:in `call'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:228:in `block in execute'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:223:in `each'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:223:in `execute'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:166:in `block in invoke_with_call_chain'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/monitor.rb:211:in `mon_synchronize'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:159:in `invoke_with_call_chain'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/task.rb:152:in `invoke'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:143:in `invoke_task'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:101:in `block (2 levels) in top_level'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:101:in `each'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:101:in `block in top_level'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:110:in `run_with_threads'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:95:in `top_level'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:73:in `block in run'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:160:in `standard_exception_handling'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/2.0.0/rake/application.rb:70:in `run'
/home/work/.rvm/rubies/ruby-2.0.0-p247/lib/ruby/gems/2.0.0/gems/rake-0.9.6/bin/rake:37:in `<top (required)>'
/home/work/.rvm/gems/ruby-2.0.0-p247/bin/rake:23:in `load'
/home/work/.rvm/gems/ruby-2.0.0-p247/bin/rake:23:in `<main>'
/home/work/.rvm/gems/ruby-2.0.0-p247/bin/ruby_executable_hooks:15:in `eval'
/home/work/.rvm/gems/ruby-2.0.0-p247/bin/ruby_executable_hooks:15:in `<main>'
Tasks: TOP => test
```

**Comment by scottdavis** [comment]
test have been broken on stable for months. and Gemfile.lock is ignored for this reason: http://yehudakatz.com/2010/12/16/clarifying-the-roles-of-the-gemspec-and-gemfile/
TL;DR Applications you checkin the lock file, Libraries you don't.

**Comment by kiafaldorius** [comment]
Eh, I agree with it in principle, but disagree with it when it comes down to library development.

So if I had code to present, I guess I'm supposed to branch off master? Tests on master seem to run ok, but I was under the impression "stable" was the branch actually released to rubygems.

When you're developing a library, the library is your app, especially when it comes down to test dependencies. You can get around it by specifying specific test gem versioning in the gemspec but you'll just introduce issues like this again anyway if a minor version change somehow breaks tests. You wouldn't ever go into a gem's directory and run bundle install. So unless bundler has some magic functionality I don't know about concerning gems, the gem's Gemfile.lock would only be used by the gem's developers anyway...

You have an unlocked "public" gem dependency list in the gemspec for people who want to run gem tests to make sure it works for their setup, and a version locked Gemfile.lock to ensure all gem devs get the same versions. But eh, no point in arguing. (heh, one of the reasons is why I don't bother keeping up to date with development blogs anymore...)

**Comment by scottdavis** [comment]
All further changes need to be made against master. the internals have changed to much

**Comment by kiafaldorius** [comment]
understood.
